config/config.go

```diff
@@ -92,6 +113,71 @@ func Unmarshall(flowConfig *FlowConfig) error {
 	return nil
 }
 
+// Print prints current configuration keys and values.
```

Comment:
Regarding "prints current configuration keys and values", the documentation does not explicitly show any print statements. Instead, it stores key-value pairs into a logger (info). This might be slightly confusing for someone reading the comment and expecting direct print output. To improve the structure of the `Print` function, I'd recommend renaming it to better reflect its purpose and improving its documentation. I'd also recommend passing the necessary configurations as parameters instead of relying on the global state.

Suggestion: 
```go
// LogConfig logs configuration keys and values if they were overridden with a config file.
// It also returns a map of keys for which the values were set by a config file.
//
// Parameters:
//     - logger: *zerolog.Event to which the configuration keys and values will be logged.
//     - flags: *pflag.FlagSet containing the set flags.
//     - conf: configuration store (assumed to be of a type that supports AllKeys and Get methods).
//
// Returns:
//     - map[string]struct{}: map of keys for which the values were set by a config file.
func LogConfig(logger *zerolog.Event, flags *pflag.FlagSet, conf interface{ AllKeys() []string; Get(key string) interface{} }) map[string]struct{} {
	keysToAvoid := make(map[string]struct{})

	if flags.Lookup(configFilePath).Changed {
		for _, key := range conf.AllKeys() {
			logger.Str(key, fmt.Sprintf("%v", conf.Get(key)))
			parts := strings.Split(key, ".")
			if len(parts) == 2 {
				keysToAvoid[parts[1]] = struct{}{}
			} else {
				keysToAvoid[key] = struct{}{}
			}
		}
	}

	return keysToAvoid
}
```

Reply:
https://github.com/onflow/flow-go/pull/4340/commits/320c424cb08961c69424af4fac139f5cd362d970